libvirt/resource_libvirt_volume.go

```diff
@@ -210,6 +229,16 @@ func resourceLibvirtVolumeCreate(d *schema.ResourceData, meta interface{}) error
 			return fmt.Errorf("can't get name for base image '%s'", baseVolumeName)
 		}
 		volumeDef.BackingStore.Path = baseVolPath
+		backingStoreXml, err := baseVolume.GetXMLDesc(0)
+		if err != nil {
+			return fmt.Errorf("can't get definitions for base image '%s'", baseVolumeName)
+		}
+		backingStore, err := newDefVolumeFromXML(backingStoreXml)
+		if err != nil {
+			return fmt.Errorf("can't parse definitions for base image '%s'", baseVolumeName)
+		}
+		log.Printf("[DEBUG] base volume format is %s", backingStore.Target.Format.Type)
+		volumeDef.BackingStore.Format.Type = backingStore.Target.Format.Type
```

Comment:
This is a repetition of the block you introduced above. Can you DRY the code by creating a dedicated function?